app/models/manageiq/providers/openstack/inventory/collector/target_collection.rb

```diff
@@ -287,6 +298,11 @@ def infer_related_vm_ems_refs_api!
       add_simple_target!(:cloud_tenants, vm.tenant_id)
       add_simple_target!(:flavors, vm.flavor["id"])
 
+      # pull the attachments from the raw attribute to avoid Fog making an unnecessary call
+      # to inflate the volumes before we need them
+      vm.attributes.fetch('os-extended-volumes:volumes_attached', []).each do |attachment|
+        add_simple_target!(:cloud_volumes, attachment["id"])
```

Comment:
@Ladas I'm having trouble getting the targeted collector to pick up the Cloud Volumes. They're added as simple targets here and I can see in my tests that this part works, but the collector method on line 194 never gets called, nor does the parser. I'm not sure what I've done differently that is causing these volume targets to not get parsed, so if you could take a look when you have time that would be much appreciated. I have a feeling I've just been staring at this too long and am simply missing something obvious. :)

Reply:
@mansam responded https://github.com/ManageIQ/manageiq-providers-openstack/pull/194/files#r161680537